File: credit-engine-api/src/main/java/com/credit/engine/application/dto/settlement/SettlementRequest.java
package com.credit.engine.application.dto.settlement;

import jakarta.validation.constraints.FutureOrPresent;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

public record SettlementRequest(
        @NotNull(message = "Informe o cedente.")
        UUID assignorId,

        @NotNull(message = "A data de liquidação é obrigatória.")
        @FutureOrPresent(message = "A data de liquidação não pode ser retroativa.")
        LocalDate valuationDate,

        @NotNull(message = "A moeda alvo de pagamento é obrigatória.")
        @Size(min = 3, max = 3, message = "O código da moeda alvo deve possuir exatamente 3 letras (ex: BRL, USD).")
        String targetCurrencyCode,

        @NotEmpty(message = "Selecione pelo menos um recebível para liquidar.")
        List<UUID> receivableIds
) {
}

File: credit-engine-ui/src/services/settlement/dto/SettlementRequest.ts
import ValidationUtils from "../../../utils/ValidationUtils.ts";

export default interface SettlementRequest {
    assignorId: string;
    valuationDate: Date;
    targetCurrencyCode: string;
    receivableIds: string[];
}

export const validateSettlementRequest = (request: SettlementRequest): { message: string }[] => {
    const errors: { message: string }[] = [];

    if (ValidationUtils.isBlank(request.assignorId))
        errors.push({ message: 'Cedente inválido.' });

    if (request.receivableIds.length === 0) {
        errors.push({ message: 'Selecione ao menos um recebível para liquidar.' });
    }

    if (!request.valuationDate) {
        errors.push({ message: 'A data de referência da liquidação e obrigatória.' });
    }

    if (ValidationUtils.isBlank(request.targetCurrencyCode)) {
        errors.push({ message: 'Selecione a moeda de liquidação.' });
    }

    return errors;
};